src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -109,12 +109,10 @@ private abstract class CompilationTrackerState
                         documentsAreFinal: false));
 
                 /// <summary>
-                /// The best compilation that is available that source generators have not ran on. May be an in-progress,
-                /// full declaration,  a final compilation, or <see langword="null"/>.
-                /// The value is an <see cref="Optional{Compilation}"/> to represent the
-                /// possibility of the compilation already having been garabage collected.
+                /// The best compilation that is available that source generators have not ran on. May be an
+                /// in-progress, full declaration,  a final compilation, or <see langword="null"/>.
                 /// </summary>
-                public ValueSource<Optional<Compilation>>? CompilationWithoutGeneratedDocuments { get; }
+                public Compilation? CompilationWithoutGeneratedDocuments { get; }
```

Comment:
after the first commit, everything that passed along a value to this was a ConstantValueSource.  So we can just remove the indirection.